lib/ASTGen/Sources/ASTGen/PluginHost.swift

```diff
@@ -24,10 +24,18 @@ enum PluginError: Error {
 
 @_cdecl("swift_ASTGen_initializePlugin")
 public func _initializePlugin(
-  opaqueHandle: UnsafeMutableRawPointer
-) {
+  opaqueHandle: UnsafeMutableRawPointer,
+  cxxDiagnosticEngine: UnsafeMutablePointer<UInt8>?
+) -> Bool {
   let plugin = CompilerPlugin(opaqueHandle: opaqueHandle)
-  plugin.initialize()
+  do {
+    try plugin.initialize()
+    return true
+  } catch {
+    // Diagnostics are emitted in the caller.
```

Comment:
If we're passing in `cxxDiagnosticEngine` now, couldn't we just emit it here?

Reply:
We can, but there's no context information such as macro name, plugin path, etc. So the diagnostics here would be less useful.

Reply:
Could pass that down 🤷. Seems like at some point we either need to pass down a string that we set here (for "what happened"), or pass down the path that we then use here to emit (or some description of the plugin).

Is passing the diagnostic engine left over?

Reply:
It was intentional leftover. I can remove it if you prefer to do so.
Or, if we really want to diagnose it here, instead of passing it down, we probably should make a `CompilerPlugin` API to get it. But the information we have here is not really useful as well it's just "failed to receive the response".

Reply:
Up to you, it doesn't seem super useful to me if we're not using it (we can always add it back later).

Reply:
I went ahead and fixed the `FIXME` in this PR. Added `CompilerPlugin.executableFilePath` property.